# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTable.cpp
# Commit: 546d74acfb81
# Full Hash: 546d74acfb81497dc00809fd4e4096edfe857ef5
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Add non-nullable references. r=lth
#   
#   This commit adds decoding, validation, and JS-API type-checking for
#   non-nullable references.
#   
# ==============================================================================

diff -r e055945526a2 -r 546d74acfb81 js/src/wasm/WasmTable.cpp
--- a/js/src/wasm/WasmTable.cpp	Tue Aug 18 16:58:55 2020 +0000
+++ b/js/src/wasm/WasmTable.cpp	Tue Aug 18 16:58:59 2020 +0000
@@ -56,6 +56,9 @@
 /* static */
 SharedTable Table::create(JSContext* cx, const TableDesc& desc,
                           HandleWasmTableObject maybeObject) {
+  // We don't support non-nullable references in tables yet.
+  MOZ_RELEASE_ASSERT(desc.elemType.isNullable());
+
   switch (desc.elemType.tableRepr()) {
     case TableRepr::Func: {
       UniqueFuncRefArray functions(